ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -219,14 +222,35 @@ public boolean accept(File dir, String name) {
   }
 
   /**
+   * Loads all index segment files that refer to segments in the log.
+   * @return all index segment files that refer to segments in the log.
+   * @throws StoreException if index segment files could not be listed from the directory.
+   */
+  private List<File> getIndexFiles() throws StoreException {
+    List<File> indexFiles = new ArrayList<>();
+    LogSegment logSegment = log.getFirstSegment();
```

Comment:
I think an iterator has limited application and this should do for now. 

For the other part of your question, in this particular case this function is called only on startup and there is no concurrency or requirements to have a point-in-time view.